sdk/python/src/dagger/engine/cli.py

```diff
@@ -30,7 +30,10 @@ def __enter__(self) -> ConnectParams:
                 stack.push(proc)
                 conn = self._get_conn(proc)
             except Exception as e:
-                raise ProvisionError("Dagger engine failed to start") from e
+                raise ProvisionError(
+                    "Dagger engine failed to start."
+                    + "If engine was manually stopped, run it again or delete it"
```

Comment:
Take care when concatenating sentences, you're missing a space between them. Also missing a comma in the last one. It's the same in all SDKs.
```suggestion
                    "Dagger engine failed to start. "
                    "If engine was manually stopped, run it again or delete it."
```